src/sounds/decoders/symphonia.rs

```diff
@@ -125,8 +125,32 @@ impl SymphoniaDecoder {
             if packet.track_id() != self.track_id {
                 continue;
             }
-            // TODO: Handle errors better when awedio allows returning errors.
-            let buf_ref = self.decoder.decode(&packet).expect("TODO");
+
+            // According to the Symphonia, some errors are indeed recoverable:
+            let buf_ref = match self.decoder.decode(&packet) {
+                Ok(buf_ref) => buf_ref,
+                // Recoverable, but this packet is void. Expect weird noises!
+                Err(Error::IoError(err)) => {
+                    if err.kind() == std::io::ErrorKind::UnexpectedEof
+                        && err.to_string() == "end of stream"
+                    {
+                        // According to Symphonia this is the only way to detect an end of stream
+                        return Err(Error::IoError(err));
+                    } else {
+                        log::warn!("IoError while decoding stream: {}", err);
```

Comment:
I am less confident that IoError should be skipped. Do you need them skipped for your project? If so do you have an example of an IoError that is skippable? If not I would prefer to still bubble these up as errors. Its always possible for the consumer to explicitly ignore IoErrors but wrapping a Sound in a Sound that calls next_sample if an IoError is returned but less flexible this way.

Reply:
I was also suprised, but it says so in the documentation [here](https://docs.rs/symphonia-core/latest/symphonia_core/codecs/trait.Decoder.html#tymethod.decode): "If a DecodeError or IoError is returned, the packet is undecodeable and should be discarded. Decoding may be continued with the next packet."

For my case, I can also live without this IoError special case, it is ugly anyways because of the ending-detection-logic. I am always decoding from RAM and thus shouldn't experience IoErrors.